### DISCUSSION TITLE
Configure VDSL, ADSL, WAN, Technicolor TG799vac (VANT-R) from Telia

### DISCUSSION THREAD

**Discussion by bufce2010** [hack-technicolor/hack-technicolor#123] [General]
Hello Again,
First of all, Big thanks to @LuKePicci  for his support and help last year to hack this model,
all this time the router have been working fine,
I also thank @FrancYescO and @Ansuel
I hope you all are safe and OK.

To follow up with my previous posts: https://github.com/hack-technicolor/hack-technicolor/issues/42 https://github.com/hack-technicolor/hack-technicolor/issues/46

~--------------------------------------------------~
**To get to the purpose of this post:**

I've successfully updated the firmware without losing root access.
![image](https://user-images.githubusercontent.com/50169186/92534307-b8bf7e00-f234-11ea-9ab2-6648f6abefc5.png)

But I've been trying to set up the router to work in two other modes:

1. First mode is  **VDSL** (to use it as the main router)
(Does VDSL configuration differs from those for ADSL? [Video for what happened](https://youtu.be/BQItl60_6oQ) )

let me quota LuKe:
> It's a standard PPPoE (plus EoA LLC encapsulation) made on top ATM interface with VPI/VCI 0/35

2.  Second mode: **WAN** (to connect an Ethernet cable from the main router to technicolor router's WAN port and use it as a wireless switch)

3. Is there a custom GUI for this model that allows me to adjust those settings normally?

Extra Info:
Router's IP address: 192.168.1.111
root password is root:root
Line actual speed : 30/3.5 Mbps (All routers all over the country wrongly reads it 40/5 Mbps).
Speed Might be upgraded to 70/7 Mbps or 70/10 Mpbs.

I appreciate it if you guided me through the process, I'll be have to use SCP for now.

Thanks in advance, for your time and efforts.

**Comment by bufce2010** [comment]
First look after custom GUI with no errors
![image](https://user-images.githubusercontent.com/50169186/92833111-c4e44080-f3d8-11ea-8cdb-3b7ca2b1bda0.png)

**Comment by bufce2010** [comment]
![image](https://user-images.githubusercontent.com/50169186/92833836-94e96d00-f3d9-11ea-8780-1401b32999be.png)

**Comment by bufce2010** [comment]
@Ansuel you might wanna add (VANT-R) to the list https://github.com/Ansuel/tch-nginx-gui
![image](https://user-images.githubusercontent.com/50169186/92834623-864f8580-f3da-11ea-8f9f-23d2a29274ee.png)

**Comment by FrancYescO** [comment]
You should be able to configure it via tch-nginx-gui, or simply editing /etc/config/network file.

Why have you set it in ADSL mode if is connected to a VDSL?!
________________________________

**Comment by bufce2010** [comment]
I think my kid broke it, he accidentally deleted all the files in /etc/config
is there a fix? 

![IMG_20200911_162330](https://user-images.githubusercontent.com/50169186/92937212-55f3ff80-f44b-11ea-9ac2-3022244c2104.jpg)

**Comment by FrancYescO** [comment]
If reset button is not working (and i suppose is not as he have not button config..) Tftp flash and start over from 0 

Another way should be the serial console.

**Comment by bufce2010** [comment]
done, tftp succeeded.
I deleted the comment as I read more carefully. 
thanks 
Edit:
Actually I needed to start from 0 as I missed up /etc/config files

**Comment by bufce2010** [comment]
> If reset button is not working (and i suppose is not as he have not button config..) Tftp flash and start over from 0

**after I'm finished with tftp, I get this error message:**
C:\Windows\system32>ssh root@192.168.1.111
ssh: connect to host 192.168.1.111 port 22: Connection refused

**AutoFlashGui says:**
Selected new variant  TG799 Telstra 17.2.0261 (Basic DDNS)
Connect attempt 1
<Response [200]>
Modem up
Authenticating
Splitting command up using semicolons
Sending command: sed -i 's#root:/bin/restricted_shell#root:/bin/ash#' /etc/passwd
Sleeping...
Sending command: echo "root:root"|chpasswd
Sleeping...
Sending command: dropbear -p 6666 &
Sleeping...
Please try a ssh connection now to 192.168.1.111 with username root and password root (change password immediately with
passwd!)  Rebooting your modem now is recommended to stop any services that have been disabled.

**Comment by bufce2010** [comment]
every time I write a comment and I go try again (while waiting for reply), then I actually succeed xD
ssh is working again
this is how I get root access using AutoFlashGui
![image](https://user-images.githubusercontent.com/50169186/92963458-c3675680-f472-11ea-9acb-6b75f23d049f.png)

**Comment by bufce2010** [comment]
would you please tell me what is wrong? [here is a video for the options from tch-nginx-gui](https://youtu.be/hTspzw9a1iY),

and this is the content of (/etc/config/network) :
```

config config 'config'
	option wan_mode 'pppoe'

config interface 'wan'
	option release '1'
	option iface6rd '6rd'
	option mtu6rd '1480'
	option vendorid 'LongLeaseTime'
	option ip4table 'main'
	option hostname 'MediaAccess TG799vac-CP1629SA5FC-17.2.0339-1021008'
	option dns_metric '0'
	option proto 'pppoe'
	option metric '0'
	option keepalive '4,20'
	option ipv6 '0'
	option peerdns '1'
	option auto '1'
	option ifname 'ptm0 vlan_data'
	option username '1077524@tedata.net.eg'
	option password '34835910'

config device 'phy_ptm0'
	option name 'ptm0'
	option mtu '1480'

config device 'phy_atm'
	option name 'atm_wan'
	option mtu '1480'

config device 'wanptm0'
	option enabled '1'
	option type '8021q'
	option name 'wanptm0'
	option vid '835'
	option ipv6 '0'

config interface 'lan'
	option type 'bridge'
	option proto 'static'
	option netmask '255.255.255.0'
	option force_link '0'
	option ip6hint '0'
	option igmp_snooping '0'
	option ipv6 '0'
	option ip6assign '64'
	option ipaddr '192.168.1.111'
	list ifname 'vlan_eth0'
	list ifname 'vlan_eth1'
	list ifname 'vlan_eth2'
	list ifname 'vlan_eth3'
	list pppoerelay 'ptm0 vlan_data'
	list pppoerelay ''

config interface 'loopback'
	option ifname 'lo'
	option proto 'static'
	option ipaddr '127.0.0.1'
	option netmask '255.0.0.0'

config globals 'globals'
	option ula_prefix 'fdf1:e6d5:40eb::/48'
	option default_ps '0'


config interface 'Guest1'
	option proto 'static'
	option ip6assign '64'
	option ipv6 '0'
	option ip6hint '1'
	option netmask '255.255.255.128'
	option ipaddr '192.168.168.126'
	option ifname 'wl0_1'
	option force_link '0'

config interface 'Guest1_5GHz'
	option proto 'static'
	option ip6assign '64'
	option ipv6 '0'
	option ip6hint '2'
	option netmask '255.255.255.128'
	option ipaddr '192.168.168.254'
	option ifname 'wl1_1'
	option force_link '0'

config device 'vlan_eth0'
	option type '8021q'
	option ifname 'eth0'
	option name 'vlan_eth0'
	option vid '1'

config device 'vlan_eth1'
	option type '8021q'
	option ifname 'eth1'
	option name 'vlan_eth1'
	option vid '1'

config device 'vlan_eth2'
	option type '8021q'
	option ifname 'eth2'
	option name 'vlan_eth2'
	option vid '1'

config device 'vlan_eth3'
	option type '8021q'
	option ifname 'eth3'
	option name 'vlan_eth3'
	option vid '1'

config device 'phy_eth3'
	option name 'eth3'
	option mtu '1500'

config switch 'bcmsw_int'
	option reset '1'
	option type 'bcmsw'
	option qosimppauseenable '0'
	option unit '0'
	option enable_vlan '0'

config switch 'bcmsw_ext'
	option reset '1'
	option type 'bcmsw'
	option qosimppauseenable '0'
	option unit '1'
	option enable_vlan '1'

config switch_vlan
	option device 'bcmsw_ext'
	option vlan '1'
	option ports '1* 2* 3* 4* 8t'

config interface 'hotspot'
	option type 'bridge'
	option ifname 'gre-hotspot'

config interface 'wan6'
	option ifname '@wan'
	option proto 'dhcpv6'
	option reqopts '23 25'
	option noslaaconly '1'
	option dns_metric '1'
	option auto '1'

config interface 'wwan'
	option auto '1'
	option metric '10'
	option proto 'mobiled'
	option session_id '0'
	option profile '1'

config interface 'iptv'
	option proto 'dhcp'
	option reqopts '1 3 6 15 33 42 51 121 249'
	option release '1'
	option iface6rd '0'
	option vendorid 'IPTV_RGW_PRIV'
	option ip4table 'iptv'
	option dnsset 'iptv'
	option ifname 'vlan_iptv'
	option auto '1'

config interface 'voip'
	option proto 'dhcp'
	option ip4table 'voip'
	option reqopts '1 3 6 15 33 42 51 121 249'
	option release '1'
	option iface6rd '0'
	option vendorid 'tel-002-'
	option dnsset 'voip'
	option ifname 'vlan_voip'
	option auto '1'

config interface 'mgmt'
	option proto 'dhcp'
	option initboot '1'
	option ip4table 'mgmt'
	option reqopts '1 3 6 15 33 42 43 51 121 249'
	option release '1'
	option iface6rd '0'
	option dnsset 'mgmt'
	option ifname 'vlan_mgmt'
	option auto '1'

config interface 'gre'
	option peeraddr '213.64.234.83'
	option mtu '1578'
	option df '1'
	option zone 'wan'
	option tunlink 'wan'
	option ttl '100'
	option proto 'gretap'
	option network 'hotspot'

config device 'phy_eth4'
	option name 'eth4'
	option mtu '1500'


config device 'vlan_data'
	option type '8021q'
	option vid '835'
	option ifname 'ptm0'
	option name 'vlan_data'

config device 'vlan_iptv'
	option type '8021q'
	option name 'vlan_iptv'
	option vid '845'
	option macaddr 'E2:B9:E5:74:49:19'
	option mtu '1500'
	option ifname 'ptm0'

config device 'vlan_voip'
	option type '8021q'
	option name 'vlan_voip'
	option vid '855'
	option macaddr 'E0:B9:E5:74:49:19'
	option mtu '1500'
	option ifname 'ptm0'

config device 'vlan_mgmt'
	option type '8021q'
	option name 'vlan_mgmt'
	option macaddr 'E2:B9:E5:74:49:18'
	option mtu '1500'
	option ifname 'ptm0'
	option vid '834'


config interface 'ipoe'
	option proto 'dhcp'
	option metric '1'
	option reqopts '1 3 6 43 51 58 59'
	option release '1'
	option neighreachabletime '1200000'
	option neighgcstaletime '2400'
	option ipv6 '1'

config device 'waneth4'
	option enabled '1'
	option type '8021q'
	option name 'waneth4'
	option vid '835'
	option ipv6 '0'

```

**Comment by FrancYescO** [comment]
`option ifname 'ptm0 vlan_data'` ??
this should be a single interface
also i see no `vlan_data` device defined.

you are probably just missing to set VLAN ID as requested by your ISP

**Comment by bufce2010** [comment]
> `option ifname 'ptm0 vlan_data'` ??
> this should be a single interface
> also i see no `vlan_data` device defined.
> 
> you are probably just missing to set VLAN ID as requested by your ISP

I've not done any thing to those options through SCP, Just cut and pasted those sections because I thought they are important and responsible for the mess I'm in.
I've edited my previous comment with the full content.
And my ISP doesn't support VLAN, 
should I reset the router and start over?

**Comment by FrancYescO** [comment]
Say exactly what have you done to have a network file like this

idk what are your isp configs but I'm pretty sure that they use a VLAN ID for the VDSL (that you should set at the bottom of broadband card)

**Comment by bufce2010** [comment]
> Say exactly what have you done to have a network file like this

Nothing xD, just cut and paste the whole config section from button to top so I can make the comment shorter when I post it here on github 

> idk what are your isp configs but I'm pretty sure that they use a VLAN ID for the VDSL (that you should set at the bottom of broadband card)

Here is a screenshot for my other router's configuration:

![image](https://user-images.githubusercontent.com/50169186/92977101-69738a80-f48c-11ea-8240-823418eddf2b.png)

**Comment by bufce2010** [comment]
I didn't modify any settings through SCP except VPI and VCI in (/etc/config/xtm) file

More explanation for what I did, 
![image](https://user-images.githubusercontent.com/50169186/92977565-9bd1b780-f48d-11ea-8230-86c986d1c454.png)

**Comment by bufce2010** [comment]
made a reset, gonna continue tomorrow (it's 1:29am here)

**Comment by FrancYescO** [comment]
If you click the edit button from the HG630 it does not show more infos? 

Try 101 as VLAN ID

**Comment by bufce2010** [comment]
> If you click the edit button from the HG630 it does not show more infos?

no, it's not checked by default, even when I apply an unlocked firmware for this Huawei router

> 
> Try 101 as VLAN ID

and what should I use for: (802.1p) , I put 0 but it didn't connect,

**Comment by FrancYescO** [comment]
didn't get what are u saying

- what happen if you click on the EDIT button in this screenshot? https://github.com/hack-technicolor/hack-technicolor/issues/123#issuecomment-691334138

- i'm saying to try using the VLAN ID 101 setting it on bottom of broadband page (0:32 of your last video); there is nothing about 802.1p.

PS. What others "connection type" are shown in the dropdown of you official modem?

**Comment by bufce2010** [comment]
> didn't get what are u saying
> 
> * what happen if you click on the EDIT button in this screenshot? [#123 (comment)](https://github.com/hack-technicolor/hack-technicolor/issues/123#issuecomment-691334138)

it's already clicked in the screenshot
this one is before clicking edit. 
![Screenshot_20200912_102131](https://user-images.githubusercontent.com/50169186/92991151-f694fe80-f4e1-11ea-8304-5b8eda757d9b.jpg)

> * i'm saying to try using the VLAN ID 101 setting it on bottom of broadband page (0:32 of your last video); there is nothing about 802.1p.

if it didn't work on the official router then it will not work on any other router, right? 

> 
> PS. What others "connection type" are shown in the dropdown of you official modem?

do you mean this? 
![Screenshot_20200912_102734](https://user-images.githubusercontent.com/50169186/92991244-d74aa100-f4e2-11ea-8698-48097faeebf0.jpg)

**Comment by bufce2010** [comment]
I followed the same settings as provided by LuKePicci in the old post, 
first I though that I've screwed settings up for the hundred time
so I got frustrated and went missing with some settings randomly via GUI:
changed SNR to: 0
changed xDSL Driver Version to: 2pv6F039k (TP-Link Archer VR900vV1)

then suddenly I got connected and internet is working!! 

I have only two things to fix:
**1- I can't access the router via DDNS host name (maybe TR069 is disabled and no option to enable it through GUI)**

2- I can't set DNS manually, router disconnects, setting it automatically displays both ISP's DNSs and manually added ones ( [Here is a video at the second 0:40 ](https://youtu.be/RwA8OpPCeaY?t=42) )
EDIT: it worked after  I've removed these lines and rebooted then reconfigured DNS settings: 
	list dns '1.1.1.1'
	list dns '8.8.8.8'
	option peerdns '0'
~-------------------------------~
settings I edited in SCP :

**/etc/config/xtm**
```
config atmdevice 'atm_wan'
	option vci '35'
	option vpi '0'
	option ulp 'eth'
	option td 'UBR'
	option enc 'llc'
	option path 'fast'

config atmdevice 'atmwan'
	option vci '35'
	option vpi '0'
	option ulp 'eth'
	option enc 'llc'
	option path 'fast'

config ptmdevice 'ptm0'
	option path 'fast'
	option priority 'low'
```

**/etc/config/network**
```
config interface 'lan'
	option type 'bridge'
	option ipaddr '192.168.1.111'
	option netmask '255.255.255.0'
	option ipv6 '0'
	option igmp_snooping '0'
	option ip6hint '0'
	option ip6assign '64'
	option force_link '0'
	option proto 'static'
	list ifname 'vlan_eth0'
	list ifname 'vlan_eth1'
	list ifname 'vlan_eth2'
	list ifname 'vlan_eth3'
	list ifname 'wanptm0'
	list pppoerelay 'atm0 vlan_data'
	list pppoerelay ''

config interface 'wan'
	option proto 'pppoe'
	option username '1077524@tedata.net.eg'
	option password '34835910'
	option authfail '0'
	option iface6rd '6rd'
	option zone6rd 'wan'
	option mtu '1492'
	option initboot '1'
	option ipv6 '0'
	option auto '1'
	list dns '1.1.1.1'
	list dns '8.8.8.8'
	option ifname 'ptm0'
	option peerdns '1'

config device 'phy_atm'
	option name 'ptm0'
	option mtu '1480'

config config 'config'
	option wan_mode 'pppoe'


config device 'wanptm0'
	option enabled '1'
	option name 'wanptm0'
	option ipv6 '0'

config device 'if0atm'
	option enabled '1'
	option name 'ptm0'

config atmdevice 'atm_wan'
	option ulp 'eth'
	option vpi '0'
	option vci '35'
	option path 'fast'
	option enc 'llc'
```

**Comment by bufce2010** [comment]
any help will be appreciated
